src/analyses/base.ml

```diff
@@ -1637,28 +1637,93 @@ struct
     in
     let inv_bin_float (a, b) fkind c op =
       let meet_bin a' b'  = FD.meet a a', FD.meet b b' in
-      let meet_com oi = (* commutative *)
-        meet_bin (oi c b) (oi c a) in
-      let meet_non oi oo = (* non-commutative *)
-        meet_bin (oi c b) (oo a c) in
       match op with
-      | PlusA  -> meet_com FD.sub
+      | PlusA  ->
+        (* A + B = C, \forall a \in A. a + b_min > pred c_min \land a + b_max < succ c_max
+            \land a + b_max > pred c_min \land a + b_min < succ c_max
+           \rightarrow A = [min(pred c_min - b_min, pred c_min - b_max), max(succ c_max - b_max, succ c_max - b_min)]
+        *)
+        let reverse_add v v' = (match FD.minimal c, FD.maximal c, FD.minimal v, FD.maximal v with 
+            | Some c_min, Some c_max, Some v_min, Some v_max when Float.is_finite (Float.pred c_min) && Float.is_finite (Float.succ c_max) -> 
+              let l = Float.min (Float.pred c_min -. v_min) (Float.pred c_min -. v_max) in 
+              let h =  Float.max (Float.succ c_max -. v_max) (Float.succ c_max -. v_min) in 
+              FD.of_interval (FD.precision c) (l, h)
+            | _ -> v') in
+        meet_bin (reverse_add b a) (reverse_add a b)
+      | MinusA ->
+        (* A - B = C \ forall a \in A. a - b_max > pred c_min \land a - b_min < succ c_max 
+            \land a - b_min > pred c_min \land a - b_max < succ c_max
+           \rightarrow A = [min(pred c_min + b_max, pred c_min + b_min), max(succ c_max + b_min, succ c_max + b_min)]
```

Comment:
Same here: `A = [pred c_min + b_min, succ c_max + b_max]` 
(+ comment here is wrong, it sais "... max(succ c_max + b_**min**, succ c_max + b_**min**) ...")